packages/application-shell/src/components/locale-switcher/locale-switcher.tsx

```diff
@@ -79,22 +121,34 @@ const LocaleSwitcher = (props: Props) => {
         name="locale-switcher"
         aria-labelledby={LOCALE_SWITCHER_LABEL_ID}
         onChange={handleSelection}
-        options={props.availableLocales.map((locale) => ({
-          label: locale,
-          value: locale,
-        }))}
+        options={localeOptions}
         components={{
           SingleValue,
           ValueContainer: PatchedValueContainer,
           MenuList: CustomMenuList,
+          GroupHeading: (groupProps) => (
+            <CustomGroupHeading {...groupProps} setIsOpen={setIsOpen} />
+          ),
         }}
         isClearable={false}
         backspaceRemovesValue={false}
         isSearchable={false}
         horizontalConstraint={'auto'}
         appearance="quiet"
         maxMenuHeight={360}
+        minMenuWidth={3}
       />
+      {/* Dialog that explains the locales */}
+      <InfoDialog
```

Comment:
Maybe we should add a test to verify the opening and closing of the dialog from the switcher.

Reply:
Yes, good point! Thank you and added here - [b717dde](https://github.com/commercetools/merchant-center-application-kit/pull/3403/commits/b717dde9db65988fba783ce2a64b3f9833d1f5d1).